src/sys/memfd.rs

```diff
@@ -35,41 +35,54 @@ libc_bitflags!(
         /// See also the hugetlb filesystem in [`memfd_create(2)`].
         ///
         /// [`memfd_create(2)`]: https://man7.org/linux/man-pages/man2/memfd_create.2.html
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGETLB;
+        /// Following only supported on OHOS.
+        /// Shift to get the huge page size.
+        #[cfg(target_env = "ohos")]
+        MFD_HUG_SHIFT;
+        /// Mask to get the huge page size.
+        #[cfg(target_env = "ohos")]
+        MFD_HUG_MASK;
+        /// hugetlb size of 64KB.
+        #[cfg(target_env = "ohos")]
+        MFD_HUG_64KB;
+        /// hugetlb size of 512KB.
+        #[cfg(target_env = "ohos")]
+        MFD_HUG_512KB;
         /// Following are to be used with [`MFD_HUGETLB`], indicating the desired hugetlb size.
         ///
         /// See also the hugetlb filesystem in [`memfd_create(2)`].
         ///
         /// [`memfd_create(2)`]: https://man7.org/linux/man-pages/man2/memfd_create.2.html
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_1MB;
         /// hugetlb size of 2MB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_2MB;
         /// hugetlb size of 8MB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_8MB;
         /// hugetlb size of 16MB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_16MB;
         /// hugetlb size of 32MB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_32MB;
         /// hugetlb size of 256MB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_256MB;
         /// hugetlb size of 512MB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_512MB;
         /// hugetlb size of 1GB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_1GB;
         /// hugetlb size of 2GB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
         MFD_HUGE_2GB;
         /// hugetlb size of 16GB.
-        #[cfg(linux_android)]
+        #[cfg(any(linux_android, target_env = "ohos"))]
```

Comment:
Same here:

> `linux_android` means `any(target_os = "linux", target_os = "android")`, and since `ohos` is a Linux libc, so it should be already covered here